internal/session/session.go

```diff
@@ -209,50 +209,51 @@ func (s *Session) Clone() interface{} {
 		case contains(opts.WithFieldMaskPaths, "HostId"):
-			return fmt.Errorf("session vet for write: host id is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "host id is immutable")
 		case contains(opts.WithFieldMaskPaths, "TargetId"):
-			return fmt.Errorf("session vet for write: target id is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "target id is immutable")
 		case contains(opts.WithFieldMaskPaths, "HostSetId"):
-			return fmt.Errorf("session vet for write: host set id is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "host set id is immutable")
 		case contains(opts.WithFieldMaskPaths, "AuthTokenId"):
-			return fmt.Errorf("session vet for write: auth token id is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "auth token id is immutable")
 		case contains(opts.WithFieldMaskPaths, "Certificate"):
-			return fmt.Errorf("session vet for write: certificate is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "certificate is immutable")
 		case contains(opts.WithFieldMaskPaths, "CreateTime"):
-			return fmt.Errorf("session vet for write: create time is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "create time is immutable")
 		case contains(opts.WithFieldMaskPaths, "UpdateTime"):
-			return fmt.Errorf("session vet for write: update time is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "update time is immutable")
 		case contains(opts.WithFieldMaskPaths, "Endpoint"):
-			return fmt.Errorf("session vet for write: endpoint is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "endpoint is immutable")
 		case contains(opts.WithFieldMaskPaths, "ExpirationTime"):
-			return fmt.Errorf("session vet for write: expiration time is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "expiration time is immutable")
 		case contains(opts.WithFieldMaskPaths, "ConnectionLimit"):
-			return fmt.Errorf("session vet for write: connection limit is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "connection limit is immutable")
 		case contains(opts.WithFieldMaskPaths, "WorkerFilter"):
-			return fmt.Errorf("session vet for write: worker filter is immutable: %w", errors.ErrInvalidParameter)
+			return errors.New(errors.InvalidParameter, op, "worker filter is immutable")
 		case contains(opts.WithFieldMaskPaths, "TerminationReason"):
 			if _, err := convertToReason(s.TerminationReason); err != nil {
-				return fmt.Errorf("session vet for write: termination reason '%s' is invalid: %w", s.TerminationReason, errors.ErrInvalidParameter)
+				return errors.Wrap(err, op)
```

Comment:
`convertToReason` is already creating the exact same error message as created here, so I just wrapped the error returned instead of making a new error here